packages/devtools-connect/package.json

```diff
@@ -0,0 +1,94 @@
+  "homepage": "https://github.com/mongodb-js/devtools-connect",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/mongodb-js/devtools-connect.git"
+  },
+  "publishConfig": {
+    "access": "public"
+  },
+  "bugs": {
+    "url": "https://github.com/mongodb-js/devtools-connect/issues"
+  },
+  "main": "dist/index.js",
+  "exports": {
+    "require": "./dist/index.js",
+    "import": "./.esm-wrapper.mjs"
+  },
+  "files": [
+    "LICENSE",
+    "dist",
+    "package.json",
+    "README.md",
+    ".esm-wrapper.mjs"
+  ],
+  "scripts": {
+    "bootstrap": "npm run compile",
+    "prepublishOnly": "npm run compile",
+    "compile": "npm run compile-ts && gen-esm-wrapper . ./.esm-wrapper.mjs",
+    "typecheck": "tsc --noEmit",
+    "eslint": "eslint src/**/*.ts",
+    "prettier": "prettier",
+    "lint": "npm run eslint . && npm run prettier -- --check .",
+    "depcheck": "depcheck",
+    "check": "npm run typecheck && npm run lint && npm run depcheck",
+    "check-ci": "npm run check",
+    "testonly": "nyc mocha --colors -r ts-node/register src/**/*.spec.ts",
+    "test": "npm run lint && npm run compile && npm run testonly",
+    "test-cov": "nyc -x \"**/*.spec.*\" --reporter=lcov --reporter=text --reporter=html npm run test",
+    "test-watch": "npm run test -- --watch",
+    "test-ci": "npm run test-cov",
+    "reformat": "npm run prettier -- --write .",
+    "prepack": "npm run compile",
+    "compile-ts": "tsc -p tsconfig.json"
+  },
+  "license": "Apache-2.0",
+  "dependencies": {
+    "@mongodb-js/oidc-http-server-pages": "0.1.0",
+    "lodash.merge": "^4.6.2",
+    "mongodb-connection-string-url": "^3.0.0",
+    "system-ca": "^1.0.2"
+  },
+  "peerDependencies": {
+    "@mongodb-js/oidc-plugin": "^0.3.1",
+    "mongodb": "^5.8.1 || ^6.0.0",
+    "mongodb-log-writer": "^1.2.0"
+  },
+  "devDependencies": {
+    "@mongodb-js/oidc-plugin": "^0.3.1",
+    "@mongodb-js/saslprep": "^1.1.4",
+    "@types/lodash.merge": "^4.6.7",
+    "@types/mocha": "^9.0.0",
+    "@types/node": "^17.0.35",
+    "@typescript-eslint/eslint-plugin": "^5.59.0",
+    "@typescript-eslint/parser": "^5.59.0",
+    "chai": "^4.3.4",
+    "eslint": "^7.25.0",
+    "eslint-config-semistandard": "^17.0.0",
+    "eslint-config-standard": "^17.1.0",
+    "eslint-plugin-import": "^2.22.0",
+    "eslint-plugin-n": "^16.6.2",
+    "eslint-plugin-promise": "^6.1.1",
+    "gen-esm-wrapper": "^1.1.0",
+    "mocha": "^8.4.0",
+    "mongodb": "^5.8.1 || ^6.0.0",
+    "mongodb-log-writer": "^1.2.0",
+    "nyc": "^15.1.0",
+    "os-dns-native": "^1.2.0",
+    "resolve-mongodb-srv": "^1.1.1",
+    "sinon-chai": "^3.7.0",
+    "ts-node": "^10.2.1",
+    "ts-sinon": "^2.0.1",
+    "typescript": "^5.0.4"
+  },
+  "optionalDependencies": {
+    "kerberos": "^2.1.0",
+    "mongodb-client-encryption": "^6.0.0",
+    "os-dns-native": "^1.2.0",
+    "resolve-mongodb-srv": "^1.1.1",
+    "socks": "^2.7.3"
```

Comment:
There are a few dependencies of the driver listed here: https://github.com/mongodb-js/devtools-connect/blob/34c8e2da3a0296c269ba8f8af44fbee0fab76549/package.json#L80-L83

Do you remember why these are being added here? I don't think it's a bad thing, just wanted to check.

(Something like `socks` could probably just be a regular dependency, it's not a native addon, it's just an optional feature of the driver but one that we want to support in all of our products)

Reply:
ah right! so devtools-connect didn't have a depcheck, and now that it was added, there were several dependencies marked missing. I added them as optional as it seemed the package was doing fine without them.

Reply:
I can move the socks, mongodb-client-encryption and kerberos are fine like this?

Reply:
Hm, I guess the depcheck here in the repo picks up on the dynamic `require()` calls 🙂 Yes, definitely fine this way